### ISSUE TITLE
Output an npz file with 20 mutation probabilities

### ISSUE THREAD

**Issue by PKU-Leroy** [dauparas/LigandMPNN#2]
Hi Justas, many thanks for your positive update!
Can ligandMPNN output an npz file similar to ProteinMPNN to obtain the mutation probability of each site?

**Comment by dauparas** [comment]
Hey! It's not available at the moment, but I can easily add this functionality if it's useful.

**Comment by PKU-Leroy** [comment]
Yes, I am interested in testing the correlation between the predicted values from ligandMPNN and experimental values. So the mutation probabilities might be very useful. I am looking forward to seeing your code update regarding this matter.



---- Replied Message ----
| From | Justas ***@***.***> |
| Date | 01/25/2024 12:43 |
| To | ***@***.***> |
| Cc | Ke-Wei ***@***.***>***@***.***> |
| Subject | Re: [dauparas/LigandMPNN] Output an npz file with 20 mutation probabilities (Issue #2) |

Hey! It's not available at the moment, but I can easily add this functionality if it's useful.

**Comment by oriondauparas** [comment]
You can get mutation probabilities by running:

```
python score.py \
        --model_type "ligand_mpnn" \
        --seed 111 \
        --single_aa_score 1\
        --pdb_path "./outputs/ligandmpnn_default/backbones/1BC8_1.pdb" \
        --out_folder "./outputs/single_aa_score_w_seq" \
        --use_sequence 1\
        --batch_size 1 \
        --number_of_batches 10
```

This script will output a dictionary per input PDB. The value you are looking for is called out_dict["mean_of_probs"] which will have probabilities for every residue and every amino acid.

**Comment by PKU-Leroy** [comment]
Thanks！



---- Replied Message ----
| From | ***@***.***> |
| Date | 02/20/2024 12:37 |
| To | ***@***.***> |
| Cc | Ke-Wei ***@***.***>***@***.***> |
| Subject | Re: [dauparas/LigandMPNN] Output an npz file with 20 mutation probabilities (Issue #2) |

You can get mutation probabilities by running:

python score.py \
        --model_type "ligand_mpnn" \
        --seed 111 \
        --single_aa_score 1\
        --pdb_path "./outputs/ligandmpnn_default/backbones/1BC8_1.pdb" \
        --out_folder "./outputs/single_aa_score_w_seq" \
        --use_sequence 1\
        --batch_size 1 \
        --number_of_batches 10


This script will output a dictionary per input PDB. The value you are looking for is called out_dict["mean_of_probs"] which will have probabilities for every residue and every amino acid.

**Comment by dauparas** [comment]
It should work now with:

```
python score.py \
        --model_type "ligand_mpnn" \
        --seed 111 \
        --single_aa_score 1\
        --pdb_path "./outputs/ligandmpnn_default/backbones/1BC8_1.pdb" \
        --out_folder "./outputs/single_aa_score_w_seq" \
        --use_sequence 1\
        --batch_size 1 \
        --number_of_batches 10
```

**Comment by dauparas** [comment]
This script will output a dictionary per input PDB. The value you are looking for is called out_dict["mean_of_probs"] which will have probabilities for every residue and every amino acid.
